### PR TITLE
Fix #3523 improved search tool

### LINKED ISSUE #106: Integrate http-proxy
For cross origin ajax calls

### LINKED ISSUE #3523: Address and coordinate search
### Description
We would like to add to the search bar in the viewer the possibility to zoom to a specific point and place a marker in that point that will be the center of the map.
There will be two possible formats: aeronautical or decimal.

This button in the search toolbar will be disabled on mobile, showing only the search for addresses

by changing the config you can:
- hide this tool (active by default only in map viewer)
- change default zoom level (default 12)
- select the search tool active (default adddres)

### PR DESCRIPTION
**Pull request by MV88**
## Description
This pr adds up to the previous #3524
image

It has:
- changed some icons in the toolbar
- added reset between switching from coordinate to addresses input fields
- moved config search services in the new hamburger menu

## Issues
 - Fix #3523

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
the search results are not reset when switching tool and search service config is under general settings

**What is the new behavior?**
those problems are gone and new ui has been implemented

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: web/client/components/mapcontrols/search/SearchBar.jsx [hunk 1]
```diff
@@ -405,13 +464,22 @@ class SearchBar extends React.Component {
...
-        }));
+        let val = isNaN(parseFloat(value)) ? "" : parseFloat(value);
+
+        this.props.onChangeCoord(coord, val);
+        if (!this.areValidCoordinates()) {
+            this.props.onClearCoordinatesSearch({owner: "search"});
+        }
     }
-    areValidCoordinates = () => (isNumber(this.state.lon) && isNumber(this.state.lat))
+    areValidCoordinates = () => (isNumber(this.props.coordinate.lon) && isNumber(this.props.coordinate.lat))
```
Comment (mbarto) [inline web/client/components/mapcontrols/search/SearchBar.jsx]: You can eventually check that coordinates are in the valid range for lat and lon (not mandatory, just a suggestion)

File: web/client/epics/search.js [hunk 2]
```diff
@@ -182,9 +182,13 @@ const zoomAndAddPointEpic = action$ =>
...
                     style: {
-                        iconGlyph: "comment",
-                        iconShape: "circle",
-                        iconColor: "green"
+                        color: '#3388ff',
+                        weight: 4,
+                        dashArray: '',
+                        fillColor: '#3388ff',
+                        fillOpacity: 0.2,
+                        iconUrl: "https://cdn.rawgit.com/pointhi/leaflet-color-markers/master/img/marker-icon-red.png",
```
Comment (mbarto) [inline web/client/epics/search.js]: can we use a local copy of this one?

File: web/client/epics/search.js [hunk 3]
```diff
@@ -182,9 +182,13 @@ const zoomAndAddPointEpic = action$ =>
...
+                        shadowUrl: "https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.3.1/images/marker-shadow.png"
```
Comment (mbarto) [inline web/client/epics/search.js]: I suppose we should already have a local copy of this

File: web/client/plugins/Search.jsx [hunk 4]
```diff
@@ -24,23 +24,39 @@ const {
...
     loading: searchState && searchState.loading,
     searchText: searchState ? searchState.searchText : "",
-    activeSearchTool: get(searchState, "activeSearchTool", "address"),
+    activeSearchTool: get(searchState, "activeSearchTool", "addressSearch"),
+    format: get(searchState, "format", "decimal"),
     selectedItems: searchState && searchState.selectedItems
 }));
 
 const SearchBar = connect(searchSelector, {
     // ONLY FOR SAMPLE - The final one will get from state and simply call textSearch
```
Comment (mbarto) [inline web/client/plugins/Search.jsx]: old comment?

File: web/client/plugins/SearchServicesConfig.jsx [hunk 5]
```diff
@@ -217,15 +215,7 @@ const SearchServicesPlugin = connect(({controls = {}, searchconfig = {}}) => ({
...
-    SearchServicesConfigPlugin: assign(SearchServicesPlugin, {
-        Settings: {
-            tool:
-                <SearchServicesButton key="searchservices"
-                text={<Message msgId="search.searchservicesbutton" />}
-            />,
-            position: 4
-        }
-    }),
+    SearchServicesConfigPlugin: assign(SearchServicesPlugin),
```
Comment (mbarto) [inline web/client/plugins/SearchServicesConfig.jsx]: remove assign

File: web/client/components/mapcontrols/search/SearchBar.jsx (hunk 1 above)
Comment (MV88) [reply web/client/components/mapcontrols/search/SearchBar.jsx]: the range validation is demanded to the coordinate editor components and is performed using enhancers (AeronauticalCoordinateEditor) or as part of the component (DecimalCoordinateEditor) and the **contraints** prop

here i'm just checking the presence of two number in order to pass the values to the zoom actions, state. etc.

Comment (mbarto) [reply web/client/components/mapcontrols/search/SearchBar.jsx]: ok, fine

File: web/client/epics/search.js (hunk 2 above)
Comment (MV88) [reply web/client/epics/search.js]: yes, i was thinking that if one overrides the resultsStyle of the Search plugin this style will be different. i'll check if we can find it from the state, (my guess is that from the config is passed to the plugin but not in the state :) )

Comment (mbarto) [reply web/client/epics/search.js]: My comment was about not depending from an external url that could disappear.

Comment (MV88) [reply web/client/epics/search.js]: i know, that's why i started with a yes... 🥇

**Pull request merged**